### PR TITLE
Fix invalid syntax in test workflow and unit tests

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Fixing the CI `test` workflow syntax.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -231,14 +231,22 @@ jobs:
         run: |
           make mockgen -j $(nproc)
           hack/tree_status.sh
-      - name: spoof crun on arm64
-        if: ${{ matrix.run.arch == "arm64" }}
-        run: cp $(which runc) /usr/bin/crun
+      - name: Download crun on arm64 # required for checkpoint restore tests
```
Comment (kwilczynski) [inline .github/workflows/test.yml:234]:
```suggestion
      - name: Download crun on arm64 # Required for checkpoint restore (criu) tests.
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -231,14 +231,22 @@ jobs:
...
+          crun --version
+        env:
+          CRUN_VERSION: 1.16.1
+          CRUN_PATH: /usr/bin/crun
       - name: Run unit tests as root
         if: ${{ matrix.run.type == 'root' }}
         run: |
           sudo PATH="$PATH" GOCACHE="$(go env GOCACHE)" GOMODCACHE="$(go env GOMODCACHE)" make testunit
-          sudo chown -R $(id -u):$(id -g) "$(go env GOCACHE)" "$(go env GOMODCACHE)"
+          sudo chown -R $(id -u):$(id -g) "$(go env GOCACHE)" "$(go env GOMODCACHE)" || true
```
Comment (kwilczynski) [inline .github/workflows/test.yml:249]: Can this fail? Just curious because of the `|| true`.

File: pkg/config/config.go [hunk 3]
```diff
@@ -895,9 +895,9 @@ func DefaultConfig() (*Config, error) {
 		RuntimeConfig: RuntimeConfig{
 			AllowedDevices:     []string{"/dev/fuse", "/dev/net/tun"},
 			DecryptionKeysPath: "/etc/crio/keys/",
-			DefaultRuntime:     defaultRuntime,
+			DefaultRuntime:     DefaultRuntime,
 			Runtimes: Runtimes{
-				defaultRuntime: defaultRuntimeHandler(),
+				DefaultRuntime: defaultRuntimeHandler(),
 			},
```
Comment (kwilczynski) [inline pkg/config/config.go:901]: Can't be private any more? How did we miss this last time... :smile:

Comment (saschagrunert) [reply pkg/config/config.go:901]: I think having it public is not too bad, isn't it?

File: .github/workflows/test.yml (hunk 2 above)
Comment (saschagrunert) [reply .github/workflows/test.yml:249]: Yes, when the modcache is not existing.

File: .github/workflows/test.yml (hunk 1 above)
Comment (saschagrunert) [reply .github/workflows/test.yml]: Fixed as suggested :+1:

File: pkg/config/config.go (hunk 3 above)
Comment (kwilczynski) [reply pkg/config/config.go:901]: Can be. :+1:

**Comment by haircommander** [comment]
/override ci/prow/ci-e2e-conmonrs

this should have no effect on that

**Pull request merged**